src/TMS/dev/TMP117.cpp

```diff
@@ -3,15 +3,17 @@
 
 namespace TMS {
 
-TMP117::TMP117(IO::I2C& i2c, uint8_t deviceID) : deviceID(deviceID), i2c(i2c) {
+TMP117::TMP117(IO::I2C* i2c, uint8_t deviceID) : deviceID(deviceID), i2c(i2c) {
     tempReg = TEMPREG;
 }
 
+TMP117::TMP117() {}
```

Comment:
```suggestion
TMP117::TMP117() : deviceID(0), i2c(nullptr) {}
```
You should initialize the member variables to default values